src/smart-components/role/add-role-permissions/schema.js

```diff
@@ -52,11 +52,28 @@ export const schemaBuilder = (container) => {
                 component: 'cost-resources',
                 name: 'cost-resources',
                 validate: [
-                  (value = []) => (value.every((p) => p.resources.length > 0) ? undefined : intl.formatMessage(messages.assignAtLeastOneResource)),
+                  (value = []) =>
+                    value.every((p) => p.resources && p.resources.length > 0) ? undefined : intl.formatMessage(messages.assignAtLeastOneResource),
```

Comment:
You can simplify this by using desctruct
```suggestion
                    value.every(({ resources }) => resources?.length > 0) ? undefined : intl.formatMessage(messages.assignAtLeastOneResource),
```